tools/compiletest/src/runtest.rs

```diff
@@ -280,15 +280,17 @@ impl<'test> TestCx<'test> {
     /// `self.testpaths.file`. An error message is printed to stdout if a result
     /// is not expected.
     fn run_kani_test(&self) {
-        self.check();
-        if self.props.kani_panic_step == Some(KaniFailStep::Check) {
-            return;
-        }
-        self.codegen();
-        if self.props.kani_panic_step == Some(KaniFailStep::Codegen) {
-            return;
+        match self.props.kani_panic_step {
+            Some(KaniFailStep::Check) => {
+                self.check();
+            }
+            Some(KaniFailStep::Codegen) => {
+                self.codegen();
+            }
+            Some(KaniFailStep::Verify) | None => {
+                self.verify();
+            }
```

Comment:
This change is necessary since we now may run `kani-compiler` from `kani` with different flags that `compiletest` wasn't picking up. This change means we avoid doing redundant work, and these differences only matter for things that expected to fail to check or codegen (which wasn't any existing tests, thankfully)

Just thought I'd note this in the review here, since it got mixed up with a bunch of other changes to get tests working again after switching to using `--harness`